bot/university/university.py

```diff
@@ -0,0 +1,378 @@
+"""Contains a Cog for all functionality regarding our University."""
+
+import xml.etree.ElementTree as ET
+import re
+from typing import Dict, List, Optional, Union
+from datetime import datetime
+import requests
+import discord
+from discord.ext import commands
+from bot.util import SelectionEmoji
+from bot.persistence import DatabaseConnector
+from bot import constants
+
+
+class UniversityCog(commands.Cog):
+    """Cog for Functions regarding our university."""
+
+    def __init__(self, bot):
+        """Initializes the Cog.
+
+        Args:
+            bot (discord.ext.commands.Bot): The bot for which this cog should be enabled.
+        """
+        self.bot = bot
+        self._db_connector = DatabaseConnector(constants.DB_FILE_PATH, constants.DB_INIT_SCRIPT)
+
+    @commands.group(name="ufind")
+    async def ufind(self, ctx: commands.Context):
+        """Command Handler for the `ufind` command.
+
+        Allows users to search the service `ufind` for various information regarding staff members, courses and exams
+        and posts them on Discord. For each of these categories exists a corresponding subcommand. If no subcommand has
+        been invoked, a help message for this command will be posted instead.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    @ufind.command(name='staff')
+    async def ufind_get_staff_data(self, ctx: commands.Context, *, search_term: str):
+        """Command Handler for the `ufind` subcommand `staff`.
+
+        Allows users to search the service `ufind` for information regarding staff members and posts them on Discord.
+        If the search result contains more than one person, an embed with reactions will be posted. The user can then
+        choose a person by reacting to said embed. If the result contains no person at all, the bot will inform the
+        user by posting a sad message.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            search_term (str): The term to be searched for (most likely firstname, lastname or both).
```

Comment:
Does it support other terms apart from name. (If not specify that it can ONLY be one of these)

Reply:
I mean, you can search for literally everything. The thing is, you probably just won't get any results back. 😜

Reply:
But the code does not break if I do so?

Reply:
Nope.